virtcontainers/kata_agent.go

```diff
@@ -1359,7 +1366,18 @@ func (k *kataAgent) handleBlockVolumes(c *Container) []*grpc.Storage {
 		}
 
 		vol.MountPoint = m.Destination
-		vol.Fstype = "bind"
+		ac, _ := c.sandbox.config.AgentConfig.(KataAgentConfig)
+		if ac.MountBlkInVM {
+			// Ensure the block device is formatted, for the devices here are specified as volumes
+			fsType, err := utils.GetDevFormat(m.Source)
+			if err != nil || fsType == "" {
+				k.Logger().WithField("device", id).WithError(err).Error("get device format failed")
+				return nil
```

Comment:
should we stop and error in this case? return an error in this case (ie, should this function return an error as well?)

Reply:
This function does need to return an error, I'm going to put it in a separate commit, and this function lacks the rollback action after error happens